rfc/techspec.md

```diff
@@ -0,0 +1,220 @@
+    payload: Any, // eg. {reason: 'failed'}
+}
+```
+
+#### Apps -> Hub
+
+**Direction**: `Hub <- App`
+
+These are messages the Hub will set event listeners for and react accordingly.
+
+- restart
+    - **id**: restart
+    - **payload**: ?
+    - **expected behavior**: **TODO**: The Hub displays any required user interface when restarting
+    
+- pause
+    - **id**: pause
+    - **payload**: None
+    - **expected behavior**: **TODO**: The app is put into a paused UI state in the Hub
+
+- **TODO**: more?
+
+### Hub -> Apps
+
+**Direction**: `Hub -> App`
+
+These are messages apps will set event listeners for and react accordingly.
+
+- resume
+    - **id**: resume
+    - **payload**: None
+    - **expected behavior**: **TODO**: App is now in focus and should resume / unpause
+
+## Client support
+
+We will not support outdated browsers (any versions of Internet Explorer).
+As this is educational content, we might as well teach the usage of up to date technology :)
+Supporting Firefox ESR is a nod to corporate, educational and enterprise environments in general which do not allow users to update their workstations. Note that running outdated software in such environments is a very relevant security risk.
+
+Desktop:
+
+* Chrome (evergreen)
+* Firefox (evergreen)
+* Firefox ESR
+* Edge (latest 2 versions)
+* Safari (latest 2 versions)
+
+Mobile (https://gs.statcounter.com/browser-market-share/mobile/worldwide):
+
+* Chrome
+* Safari
+* Samsung Internet
+* UC Browser
+
+Note that browsers such as Brave, Opera, ... are likely still implicitly supported, just not specifically targeted.
+
+For implementation we suggest the use of [browserslist](https://github.com/browserslist/browserslist) in your tech stack. 
+
+Suggested entry:
+
+```
+# file: .browserslistrc
+> 2%
+last 2 versions
+Firefox ESR
+not dead
+not ie 11
+```
+
+Example Supported output:
+
+`npx browserslist`
+
+```
+and_chr 79
+and_ff 68
+and_qq 1.2
+and_uc 12.12
+android 76
+baidu 7.12
+chrome 79
+chrome 78
+edge 18
+edge 17
+firefox 72
+firefox 71
+firefox 68
+ios_saf 13.3
+ios_saf 13.2
+ios_saf 13.0-13.1
+ios_saf 12.2-12.4
+kaios 2.5
+op_mini all
+op_mob 46
+opera 64
+opera 63
+safari 13
+safari 12.1
+samsung 10.1
+samsung 9.2
+```
+
+Browserslist can be integrated into your tech stack to automatically provide relevant polyfills via Babel, PostCSS, etc.
+
+## Screen Resolution support
+
+See: https://gs.statcounter.com/screen-resolution-stats
+
+- Mobile views should support down to 360x640
+    - We might require users to switch to portrait mode to play certain games?
+- Tablet displays could be handled as mobile device in portrait view, desktop in landscape
+- Desktop 1366x768 up to 1920x1080, while gracefully handling ultrawide screens and larger resolutions
+
+**NOTE**: The Hub can provide a query parameter to Apps to request a mobile view if this is desired.
+
+## i18n
+
+Use [ISO 639-1](https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes) codes.
+
+## Offline Support
+
```

Comment:
Addendum: in the application we wrote also "available online as well as offline by downloading it without special 1T knowledge. (pag 28 pdf document)". Will this be a problem?